### PR TITLE
waitForTransactionReceipt fix

### PR DESCRIPTION
**Pull request by jdevcs**
## Description

https://github.com/web3/web3.js/issues/6328

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

**Comment by jdevcs** [comment]
https://github.com/web3/web3.js/issues/6447 will also be fixed, 
For terminating receipt looking logic in less time, just set `web3.transactionPollingTimeout` to some lower value if you want lower thn [predefined](https://github.com/web3/web3.js/blob/b8fa7124d1d1e4c9c16b60ad2ce58559f72efd27/packages/web3-core/src/web3_config.ts#L76).

File: packages/web3-utils/src/promise_helpers.ts [hunk 1]
```diff
@@ -73,47 +73,56 @@ export async function waitWithTimeout<T>(
...
-	return polledRes;
+/**
+ * Repeatedly calls an async function with a given interval until the result of the function is defined (not undefined or null),
+ * or until a timeout is reached.
+ * pollTillDefinedAndReturnIntervalId() function should be used instead of pollTillDefined if you need IntervalId in result.
+ * This function will be deprecated in next major release so use pollTillDefinedAndReturnIntervalId(). 
+ * @param func - The function to call.
+ * @param interval - The interval in milliseconds.
+ */
+export async function pollTillDefined<T>(
```
Comment (jdevcs) [inline packages/web3-utils/src/promise_helpers.ts:120]: keeping `pollTillDefined` so we dnt break API as its exported function from utils.

File: packages/web3-utils/test/unit/promise_helpers.test.ts [hunk 2]
```diff
@@ -121,6 +122,46 @@ describe('promise helpers', () => {
...
+				}
+				return new Promise((_, reject) => {
+					reject(dummyError);
+				});
+			};
+			const [promise] = pollTillDefinedAndReturnIntervalId(asyncHelper, 100);
+			await expect(promise).rejects.toThrow(dummyError);
+		});
+	});
+
```
Comment (luu-alex) [inline packages/web3-utils/test/unit/promise_helpers.test.ts:186]: should have a testcase to manually `clearInterval` successfully

**Review by avkos — CHANGES_REQUESTED** [review]
Coverage decreased. Will be great to fix it

File: packages/web3-utils/test/unit/promise_helpers.test.ts (hunk 2 above)
Comment (jdevcs) [reply packages/web3-utils/test/unit/promise_helpers.test.ts:186]: added

**Review by avkos — APPROVED** [review]

**Review by luu-alex — APPROVED** [review]

**Pull request merged**